content/how-to/bpmn/debug-workflows.md

```diff
@@ -0,0 +1,248 @@
+You can also use debug flags and variables to trace variable context as it transforms across the workflow.
+
+## Strategies to handle errors
+
+All error handling likely uses some conditional logic.
+The workflow author anticipates the error and then writes some logic to conditionally handle it.
+However, you have many ways to handle conditions. When deciding how to direct flows, consider both the context of the error and overall readability of your diagram.
+This section describes some key strategies.
+
+### Gateways
+
+Use [exclusive gateways]({{< relref "/how-to/bpmn/bpmn-elements#gateways" >}}) for any type of error handling.
+For example, you might define a normal range for a value, then send alerts for when the value falls outside of this range.
+If it makes sense, these error branches also might flow into an early end event.
+
+<!-- vale off -->
+{{< bigFigure
+src="/images/bpmn/screenshot-rhize-bpmn-error-handling-custom-response.png"
+alt="A BPMN workflow with customResponse in the output of the end node"
+caption="Download this workflow from [BPMN templates](https://github.com/libremfg/bpmn-templates/tree/main/custom-response-error-events)"
+width="80%"
+>}}
+<!-- vale on -->
+
+### JSON schema validation
+
+Validation can greatly limit the scope of possible errors.
+To validate your JSON payloads, use the [JSON schema task]({{< relref "/how-to/bpmn/bpmn-elements#json-schema" >}}).
+
+The JSON schema task outputs a boolean value that indicates whether the input conforms to the schema that you set.
+You can then set a condition based on whether this `valid` variable is true, and create logic to handle errors accordingly.
+For example, this schema requires that the input variables include a property `arr` whose
+value  is an array of numbers.
+
+
+```json
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Generated schema for Root",
+  "type": "object",
+  "properties": { "arr": { "type": "array", "items": { "type": "number" } } },
+  "required": ["arr"]
+}
+```
+
+In a production workflow, you might use this exact schema to validate the input for a function that calculates statistics (perhaps choosing a different variable name).
+
+
+{{< bigFigure
+src="/images/bpmn/screenshot-rhize-bpmn-json-schema.png"
+alt="Screenshot of a conditional that branches when the JSON schema task receives invalid input."
+caption="A conditional that branches when the JSON schema task receives invalid input. [Download the template](https://github.com/libremfg/bpmn-templates/tree/main/call-activity-calculate-stats)"
+width="35%"
+>}}
+
+
+
+
+### JSONata conditions
+
+Besides the logical gateway, it may make sense to use JSONata ternary expressions in one of the many parameters that accepts JSONata expressions.
+For example, this expression creates one message body if `valid` is `true` and another if not:
+
+```jsonata
+=
+{
+  "message": $.valid ? "Schema is valid" : "Invalid schema" 
```

Comment:
This is also pedantic, but is the schema valid/invalid or the variable context? It looks to me like the schema is valid, but the variable context is invalid against the schema.

Reply:
Love these details :+1: